ui/src/main.mjs

```diff
@@ -71,20 +74,69 @@ fetch('_setup')
 
         store.commit('setup/set', setup)
 
-        const socket = io(setup.socketio)
-
-        // handle final disconnection
-        socket.on('disconnect', (reason) => {
-            console.log('SIO disconnected', reason)
-        })
-
-        socket.on('connect', () => {
-            console.log('SIO connected')
-        })
+        let disconnected = false
+        let disconnectedAt = null
+
+        const MAX_TIMER = 300000 // 5 minutes
+
+        let socket = null
+
+        function connect () {
+            socket = io(setup.socketio)
+
+            // handle final disconnection
+            socket.on('disconnect', (reason) => {
+                disconnected = true
+                disconnectedAt = new Date()
+                // tell the user we're trying to connect
+                Alerts.emit('Connection Lost', 'Attempting to reconnect to server...', 'red', {
+                    displayTime: 0,
+                    allowDismiss: false,
+                    showCountdown: false
+                })
+                socket.destroy()
+                // attempt to reconnect
+                reconnect()
+            })
+
+            socket.on('connect', () => {
+                console.log('SIO connected')
+                // if we've just disconnected (i.e. aren't connecting for the first time)
+                if (disconnected) {
+                    // send a notification/alert to the user to let them know the connection is live again
+                    Alerts.emit('Connected', 'Connection re-established.', '#1BC318', {
+                        displayTime: 3,
+                        allowDismiss: true,
+                        showCountdown: true
+                    })
+                }
+                disconnected = false
```

Comment:
Recommendation: Here, we should probably call `clearTimeout(...)` to halt any ongoing timeout. Not mandatory since you do test the `disconnected` boolean. (good practice)

Reply:
I shall make it so, thanks Steve